### PR TITLE
net/lwip: fix sem_tickwait error handleing

### PR DESCRIPTION
**Pull request by pillip8282**
sem_tickwait API was changed however lwIP didn't apply changed API.

### CODE REVIEW

File: os/net/lwip/sys/arch/sys_arch.c [hunk 1]
```diff
@@ -483,21 +482,22 @@ u32_t sys_arch_sem_wait(sys_sem_t *sem, u32_t timeout)
...
-			} else if (status == -ETIMEDOUT) {
+			int err_no = get_errno();
+			if (err_no == ETIMEDOUT) {
 				return SYS_ARCH_TIMEOUT;
-			} else {
+			} else if (err_no == ENOMEM) {
+				LWIP_ASSERT("status == ENOMEM", 0);
+			} else if (err_no == ECANCELED) {
+				return SYS_ARCH_CANCELED;
+			}else {
```
Comment (sunghan-chang) [inline os/net/lwip/sys/arch/sys_arch.c:492]: } else {
Could you add a space between `} else`?

Comment (jeongchanKim) [inline os/net/lwip/sys/arch/sys_arch.c:492]: Please add space after `{`.

Comment (pillip8282) [reply os/net/lwip/sys/arch/sys_arch.c:492]: Thanks I'll modify it.

Comment (pillip8282) [reply os/net/lwip/sys/arch/sys_arch.c:492]: Thanks I'll add a space

**Review by an4967 — APPROVED** [review]
LGTM

**Pull request merged**